korman/nodes/node_avatar.py

```diff
@@ -71,3 +72,395 @@ def export(self, exporter, bo, so):
     @property
     def requires_actor(self):
         return True
+
+
+class PlasmaAnimStageOrderSocket(PlasmaNodeSocketBase, bpy.types.NodeSocket):
+    bl_color = (0.412, 0.2, 0.055, 1.0)
+
+
+class PlasmaAnimStageAdvanceSocketIn(PlasmaAnimStageOrderSocket):
+    auto_advance = BoolProperty(name="Advance to Next Stage",
+                                description="Automatically advance to the next stage when the animation completes instead of halting",
+                                default=True)
+
+    def draw(self, context, layout, node, text):
+        if not self.is_linked:
+            layout.prop(self, "auto_advance")
+        else:
+            if self.links[0].from_node.stage_id is not None:
+                layout.label("{} {}".format(text, self.links[0].from_node.stage_id))
+            else:
+                layout.label(text)
+
+
+class PlasmaAnimStageRegressSocketIn(PlasmaAnimStageOrderSocket):
+    auto_regress = BoolProperty(name="Regress to Previous Stage",
+                                description="Automatically regress to the previous stage when the animation completes instead of halting",
+                                default=True)
+
+    def draw(self, context, layout, node, text):
+        if not self.is_linked:
+            layout.prop(self, "auto_regress")
+        else:
+            if self.links[0].from_node.stage_id is not None:
+                layout.label("{} {}".format(text, self.links[0].from_node.stage_id))
+            else:
+                layout.label(text)
+
+
+class PlasmaAnimStageOrderSocketOut(PlasmaAnimStageOrderSocket):
```

Comment:
Best to have any classes used by blender inherit the blender base directly. Meaning this should inherit from `bpy.types.NodeSocket` explicitly, instead of implicitly via `PlasmaAnimStageOrderSocket`. Interesting issues occur, mainly in properties, if you expect for inheritance to work properly with Blender classes.

Reply:
That's unfortunate.